tests/api_connexion/test_webserver_auth.py

```diff
@@ -0,0 +1,129 @@
+        self.tester = self.appbuilder.sm.find_user(username="test")
+        if not self.tester:
+            self.appbuilder.sm.add_user(
+                username="test",
+                first_name="test",
+                last_name="test",
+                email="test@fab.org",
+                role=role_admin,
+                password="test",
+            )
+        clear_db_pools()
+
+    def tearDown(self) -> None:
+        clear_db_pools()
+
+    def test_successful_login(self):
+        token = "Basic " + b64encode(b"test:test").decode()
+        with self.app.test_client() as test_client:
+            response = test_client.post("api/v1/login", headers={"Authorization": token})
+        assert isinstance(response.json["token"], str)
+        assert response.json["user"]['email'] == "test@fab.org"
+
+    def test_can_view_other_endpoints(self):
+        token = "Basic " + b64encode(b"test:test").decode()
+        with self.app.test_client() as test_client:
+            response = test_client.post("api/v1/login", headers={"Authorization": token})
+            assert current_user.email == "test@fab.org"
+            token = response.json["token"]
+            response2 = test_client.get("/api/v1/pools", headers={"Authorization": "Bearer " + token})
+        assert response2.status_code == 200
+        assert response2.json == {
+            "pools": [
+                {
+                    "name": "default_pool",
+                    "slots": 128,
+                    "occupied_slots": 0,
+                    "running_slots": 0,
+                    "queued_slots": 0,
+                    "open_slots": 128,
+                },
+            ],
+            "total_entries": 1,
+        }
+
+    def test_raises_for_the_none_algorithm(self):
+        token = "Basic " + b64encode(b"test:test").decode()
+        payload = {
+            'exp': datetime.utcnow() + timedelta(minutes=10),
+            'iat': datetime.utcnow(),
+            'sub': self.tester.id,
+        }
+        forgedtoken = jwt.encode(payload, key=None, algorithm=None).decode()
+        with self.app.test_client() as test_client:
+            test_client.post("api/v1/login", headers={"Authorization": token})
+            assert current_user.email == "test@fab.org"
+            response = test_client.get("/api/v1/pools", headers={"Authorization": "Bearer " + forgedtoken})
+        assert response.status_code == 401
+
+    @parameterized.expand(
+        [
+            ("basic",),
+            ("basic ",),
+            ("bearer",),
+            ("test:test",),
+            (b64encode(b"test:test").decode(),),
+            ("bearer ",),
+            ("basic: ",),
+            ("basic 123",),
+        ]
+    )
+    def test_malformed_headers(self, token):
+        with self.app.test_client() as test_client:
+            response = test_client.get("/api/v1/pools", headers={"Authorization": token})
+            assert response.status_code == 401
```

Comment:
Is 400 (Client Error) more appropriate here?

Reply:
Yes, it's more appropriate but because of auth_backends, every error is returned as 401 in check_authorization